HDDS-11017. Migrated to ECharts, Vite and AntD v4 with eslint, prettier

devabhishekpal opened the pull request:
## What changes were proposed in this pull request?

- While conducting a fossa scan we found that node-notifier and jsdom is not compatible with Apache license
- This PR migrates from the react-scripts used in the current project to Vite and Vitest which resolved the licensing issue
- Apart from that this will bring additional improvements like:
  - Faster local development time
  - Faster builds
  - Migration to AntD v5 which will allow us to directly upgrade to newer versions
  - Shift from xo to eslint and prettier - this will make the development experience more smooth and consistent for contributors

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-11017

## How was this patch tested?
The patch was tested manually on local build
<img width="648" alt="Overview Page" src="https://github.com/apache/ozone/assets/43001336/0ce5f527-55d7-4fcc-bfda-32bcc8a05190">
<img width="648" alt="Volumes Page" src="https://github.com/apache/ozone/assets/43001336/adb052d9-9427-48ce-8d01-381cc9de1e8c">
<img width="648" alt="Volumes with ACL drawer" src="https://github.com/apache/ozone/assets/43001336/5f343e14-fb2d-4cbd-84bd-cb733cff474d">
<img width="648" alt="Buckets Page" src="https://github.com/apache/ozone/assets/43001336/1332ea28-47ff-4b02-a63d-7c3f50f74fe1">
<img width="648" alt="Buckets with ACL Drawer" src="https://github.com/apache/ozone/assets/43001336/08c78cc5-ad0a-46dc-a833-01e7bfb330e9">
<img width="648" alt="Datanodes Page" src="https://github.com/apache/ozone/assets/43001336/73591442-d8b8-42e3-a957-edabd5666bb1">
<img width="648" alt="Pipelines Page" src="https://github.com/apache/ozone/assets/43001336/a7b72daa-099f-4af5-96c8-7b8adab837a6">
<img width="648" alt="Containers Page" src="https://github.com/apache/ozone/assets/43001336/5c50c6d9-ee5d-4f3e-bb6d-b8d2f57d4a39">
<img width="648" alt="Insights File Size dist page" src="https://github.com/apache/ozone/assets/43001336/e683d97c-fb5b-42f3-8698-47cef3139933">
<img width="648" alt="File Size Dist full chart" src="https://github.com/apache/ozone/assets/43001336/8e0bc59c-6736-42d7-b1a9-d8797e13aba9">
<img width="648" alt="Ingihts Container Size Distribution" src="https://github.com/apache/ozone/assets/43001336/f7a8074d-dd50-42d6-a3ba-4448613e091e">
<img width="648" alt="OM DB Insights Page" src="https://github.com/apache/ozone/assets/43001336/02429628-4160-4b83-b395-dcae92573456">
<img width="648" alt="DU Page" src="https://github.com/apache/ozone/assets/43001336/ddc46a31-8c75-4e52-91bf-6cff9c7330dd">
<img width="648" alt="Heatmap default path" src="https://github.com/apache/ozone/assets/43001336/5ac4e7f3-92ac-4640-a5ee-19759ed35059">
<img width="648" alt="Heatmap with buckets and changed path" src="https://github.com/apache/ozone/assets/43001336/13d29838-076b-409e-98c9-91d39e6e7501">

devabhishekpal commented:
Currently library selection no longer has any licensing issues.

devmadhuu commented on hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/app.tsx:
```diff
@@ -38,40 +39,55 @@ interface IAppState {
+      //     controlItemBgHover: 'rgba(29, 165, 122, 0.12)'
+
+      //   }
+      // }}>
```
Can we remove this commented code ?

devabhishekpal replied:
This has been removed in the latest commit.
Thanks

smitajoshi12 commented:
@devabhishekpal 
Can we update node and pnpm version also?

devmadhuu reviewed (approved):
Thanks @devabhishekpal for this effort and patch. Changes LGTM+1. Just a nit from my side.Request @dombizita to have a look.

devabhishekpal commented:
Hi @smitajoshi12 , so for Node v18 - it no longer supports CentOS 7 and older linux distros.
We would need to then specify it that the project cannot be built on older linux dist.
Hence we are stuck with Node v16 for the time being.
pnpm v9 required Node 18. So I am upgrading to the latest supported pnpm v8.15.7.

smitajoshi12 commented:
@devabhishekpal 
In Disk Usage CSS aligmnet has been changed for Show Metata Data button.

![image](https://github.com/apache/ozone/assets/112169209/8dcd00d2-50d3-4c24-aaed-69c6a89e0c9f)

ivandika3 commented on hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/package.json:
```diff
@@ -63,109 +56,23 @@
   },
   "devDependencies": {
-    "@types/react-plotly.js": "^2.2.4",
+    "@types/react": "16.8.15",
+    "@types/react-dom": "16.8.4",
+    "@types/react-router-dom": "^4.3.5",
```
Does React dependencies supposed to be moved from `dependencies` to `devDependencies`?

devabhishekpal replied:
Yes Ivan, so `@types/...` provide typedefs.
We should include them in the devDependencies only if we are exporting those types in the final build as well.
For example if we are creating a custom library, and we need to provide the types to our end users to use the libs then it should mandatorily be in dependencies.

But in this case since we are only consuming the types as a part of the development and not required for the final build product we can safely move them out I believe.

ivandika3 commented on hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/components/aclDrawer/aclDrawer.tsx:
```diff
@@ -70,7 +71,6 @@ export class AclPanel extends React.Component<IAclDrawerProps> {
           closable={false}
           visible={visible}
           getContainer={false}
-          style={{position: 'absolute'}}
```
Is this removed due to the eslint?

devabhishekpal replied:
Hi, I might have removed it as styles were not supported in AntD v5 in this manner.
Thanks for pointing this out. Added the style back in the latest commit.

ivandika3 replied:
I see, we can remove it anyway if it's not used in the AntD v5. Or you can file a follow up ticket. Just making sure that the all the changes are related to the dependency migrations.

smitajoshi12 commented:
@devabhishekpal 

Can we move pi chart alignment to left side. 

![image](https://github.com/apache/ozone/assets/112169209/069f808d-2ee8-47b3-9b43-9893a2b4c29f)

ivandika3 commented on hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/app.less:
```diff
@@ -34,7 +35,7 @@
 
 .page-header {
-  padding: 10px 0;
+  padding: 20px 0;
```
Any reason of changing this? If it's not related to the dependency migration let's address it in follow up ticket.

devabhishekpal replied:
Sure, thanks Ivan. It was just changed to add a bit more spacing in the heading element.
But even the previous change works. Reverting in the latest commit

smitajoshi12 commented:
@devabhishekpal 

Css Issue in heatmap when selecting dropdown

![image](https://github.com/apache/ozone/assets/112169209/5d6e9754-3abe-4943-afd4-02c69a342559)

devabhishekpal commented:
Hi @smitajoshi12, for https://github.com/apache/ozone/pull/6841#issuecomment-2188075777, if we move the pie-chart to the left side it would look odd as without the metadata panel open, it would have lots of empty space on the right hand side.
So we are keeping it in the center as when the metadata is viewed the user will either way not be interacting with the chart.

Please do let me know if you want to shift to the left or if you have any other alternate inputs.
Thanks

smitajoshi12 commented:
> Hi @smitajoshi12, for [#6841 (comment)](https://github.com/apache/ozone/pull/6841#issuecomment-2188075777), if we move the pie-chart to the left side it would look odd as without the metadata panel open, it would have lots of empty space on the right hand side. So we are keeping it in the center as when the metadata is viewed the user will either way not be interacting with the chart.
> 
> Please do let me know if you want to shift to the left or if you have any other alternate inputs. Thanks

@ivandika3  @dombizita 
Can suggest but if  entity name is lengthy then meta data summary pop is overlapping or we can mimimize metadata summary pop up so we will get clear visibility of entity names.

devabhishekpal commented:
@smitajoshi12 could you let me know the screen resolution you are using for https://github.com/apache/ozone/pull/6841#issuecomment-2188080629?
It seems in my resolution this issue is not present.

https://github.com/apache/ozone/assets/43001336/493257b8-712a-4bc9-b1bb-0fa8fe4e40fc

It is good to have other resolutions to test as well. Thanks a lot for bringing this to my attention.

devabhishekpal commented:
> > Hi @smitajoshi12, for [#6841 (comment)](https://github.com/apache/ozone/pull/6841#issuecomment-2188075777), if we move the pie-chart to the left side it would look odd as without the metadata panel open, it would have lots of empty space on the right hand side. So we are keeping it in the center as when the metadata is viewed the user will either way not be interacting with the chart.
> > Please do let me know if you want to shift to the left or if you have any other alternate inputs. Thanks
> 
> @ivandika3 @dombizita Can suggest but if entity name is lengthy then meta data summary pop is overlapping or we can mimimize metadata summary pop up so we will get clear visibility of entity names.

So I changed the current legend to go on the left hand side for better visibility.
After my current changes it looks like below:

<img width="1728" alt="Screenshot 2024-06-25 at 12 45 45" src="https://github.com/apache/ozone/assets/43001336/d563dd0b-5de3-4826-858a-1766c5e07c1f">

smitajoshi12 commented:
> > > Hi @smitajoshi12, for [#6841 (comment)](https://github.com/apache/ozone/pull/6841#issuecomment-2188075777), if we move the pie-chart to the left side it would look odd as without the metadata panel open, it would have lots of empty space on the right hand side. So we are keeping it in the center as when the metadata is viewed the user will either way not be interacting with the chart.
> > > Please do let me know if you want to shift to the left or if you have any other alternate inputs. Thanks
> > 
> > 
> > @ivandika3 @dombizita Can suggest but if entity name is lengthy then meta data summary pop is overlapping or we can mimimize metadata summary pop up so we will get clear visibility of entity names.
> 
> So I changed the current legend to go on the left hand side for better visibility. After my current changes it looks like below:
> 
> <img alt="Screenshot 2024-06-25 at 12 45 45" width="1728" src="https://private-user-images.githubusercontent.com/43001336/342620787-d563dd0b-5de3-4826-858a-1766c5e07c1f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTkzMDE2NTcsIm5iZiI6MTcxOTMwMTM1NywicGF0aCI6Ii80MzAwMTMzNi8zNDI2MjA3ODctZDU2M2RkMGItNWRlMy00ODI2LTg1OGEtMTc2NmM1ZTA3YzFmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA2MjUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNjI1VDA3NDIzN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWViOTNjZmY0NDQ4NzJhYjJhNzUxMDVkMzVhZTVmY2ZkYmI5YjY3MWJkZTIwZmU0YjE3MzdjNTg0NzQ0ZDk0MDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.TPpVt-xwZFciDd8NqlCS62cv7uEweTDfrICGdXX7uO0">

@devabhishekpal 
These disk usage changes looks good to me.

devabhishekpal commented:
After adjusting for media queries in the heatmap we are having the following UI:
Resolution was reduced to adjust for Github media size

https://github.com/apache/ozone/assets/43001336/a3469127-9df4-4465-b0c8-1038548363ea

smitajoshi12 commented on hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/views/insights/om/om.tsx:
```diff
@@ -1014,59 +1046,50 @@ export class Om extends React.Component<Record<string, object>, IOmdbInsightsSta
+            </Tabs.TabPane>
+            <Tabs.TabPane key='4' tab={(
+              <label>Deleted Container Keys&nbsp;&nbsp;
+                <Tooltip placement='top' title={'Keys mapped to Containers in DELETED state SCM.'}>
```
@devabhishekpal 
It looks good now after your changes.

devabhishekpal replied:
Addressed in the latest commit, added the missing datasource

devabhishekpal commented on hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/api/db.json:
Made the change in the latest commit.
Thanks

smitajoshi12 commented:
@devabhishekpal 

while doing cluster testing  volume1 was having data  and 100% and Total Data Size: 63.1 KB
It should be inside pie chart Size and Percentage is accurate just position is changed.

![image](https://github.com/apache/ozone/assets/112169209/4937d23a-f71c-40c3-8b69-39e32b1150d0)

{
    "status": "OK",
    "path": "/",
    "size": 64614,
    "sizeWithReplica": -1,
    "subPathCount": 39,
    "subPaths": [
        {
            "key": false,
            "path": "/volume1",
            "size": 64614,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/s3v",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-0-79279",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-1-38323",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-10-54251",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-11-70290",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-12-44556",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-13-18787",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-14-46679",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-15-55529",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-16-51548",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-17-87656",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-18-09750",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-19-95360",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-2-75113",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-20-41102",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-21-75670",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-22-58601",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-23-97664",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-24-56683",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-25-42101",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-26-76437",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-27-82747",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-28-42441",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-29-31116",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-3-08351",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-30-12401",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-31-88150",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-32-90258",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        },
        {
            "key": false,
            "path": "/vol-33-64420",
            "size": 0,
            "sizeWithReplica": -1,
            "isKey": false
        }
    ],
    "sizeDirectKey": -1
}

![image](https://github.com/apache/ozone/assets/112169209/e678cf8a-2a58-4ec5-b492-ce695077e2d3)

devabhishekpal commented:
> @devabhishekpal
> 
> while doing cluster testing volume1 was having data and 100% and Total Data Size: 63.1 KB It should be inside pie chart Size and Percentage is accurate just position is changed. How we will come to know which are zero size entities and which entities are having data.
> 

Hi @smitajoshi12, so in general when we are hovering over an area the area becomes highlighted and elevated in ECharts.
Apart from that I added new changes to also display the current path element name in the tooltip in the legend color.
After changes it looks like below:
<img width="819" alt="Screenshot 2024-06-25 at 17 08 03" src="https://github.com/apache/ozone/assets/43001336/74a57ac3-90df-488d-8477-377f1cd41db7">

smitajoshi12 commented:
With Same Cluster data results are different in pi chart

filecount 
[
    {
        "volume": "volume1",
        "bucket": "bucket1",
        "fileSize": 32768,
        "count": 5
    },
    {
        "volume": "volume1",
        "bucket": "bucket2",
        "fileSize": 32768,
        "count": 3
    }
]
------------------------------
container count
[{"containerSize":536870912,"count":1}]

Old Library
![image](https://github.com/apache/ozone/assets/112169209/b63293db-b57d-40e5-9d2f-fec6795000c7)

![image](https://github.com/apache/ozone/assets/112169209/ce9035f6-c0f9-4d15-bdad-ccf1d00884c7)

New Library
![image](https://github.com/apache/ozone/assets/112169209/7bd6da24-96ec-4316-b84c-e396d2af3dca)

![image](https://github.com/apache/ozone/assets/112169209/f04c64c7-3cdd-43a1-bbbf-f598a55a8fec)


[5:41](https://cloudera.slack.com/archives/D05C3B8CR8U/p1719317511907429)
"fileSizeCounts": [
    {
        "volume": "volume1",
        "bucket": "bucket1",
        "fileSize": 32768,
        "count": 5
    },
    {
        "volume": "volume1",
        "bucket": "bucket2",
        "fileSize": 32768,
        "count": 3
    }
]

devabhishekpal commented:
Hi @smitajoshi12, thanks for pointing out this issue.
Fixed it in the latest patch.
With your data now it is looking like below:
<img width="598" alt="Screenshot 2024-06-25 at 18 00 32" src="https://github.com/apache/ozone/assets/43001336/6890e914-36a4-4325-a91e-ade9229f263b">

One change from the original is that we are referring to the range instead of the actual value as in the original it seemed like multiple entities were having the same fixed container size.
Showing the range let's the user know that it is not a fixed size but a range in which the entities are spread

smitajoshi12 commented:
@devabhishekpal 

Looks like Container ID column alignment has been changed.

![image](https://github.com/apache/ozone/assets/112169209/c1717426-7b77-4bff-ab3c-abc428900d39)

smitajoshi12 commented:
@devabhishekpal 

Can we increase width of Pi Chart align more to left side so can get visibility for all entities.

![image](https://github.com/apache/ozone/assets/112169209/6e9241d2-e6c1-4e75-a3ff-671ba153809a)

devabhishekpal commented:
> @devabhishekpal In Overview Page In Route.json entries are also missing for local run getting 404 in local.
> 
> ![image](https://private-user-images.githubusercontent.com/112169209/343683342-199a48d3-9c05-4885-becd-d7b37a67bde5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk0ODcxODMsIm5iZiI6MTcxOTQ4Njg4MywicGF0aCI6Ii8xMTIxNjkyMDkvMzQzNjgzMzQyLTE5OWE0OGQzLTljMDUtNDg4NS1iZWNkLWQ3YjM3YTY3YmRlNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyN1QxMTE0NDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00YjE0ZjljZmI0NDAwMTA2YmFmZTY2MTdhYmFkZjgyZDc4ZmZjOTQzYTk2ZjcwZTk2NmUyNzQwZmNjMzQ0NzYwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.06X0DfzXVgc9vi7qaDCC-gN2kRpObEXtDhQxxmNIcK8)

Hi @smitajoshi12 , I checked the older routes.json as well. I mean from older branch (Ozone 1.4.0) it seems the path mapping to `/triggerdbsync/om` was never present, hence the 404 response.
But if we test it on a cluster instead of in dev mode, the API is being called. So it is okay in built files.

smitajoshi12 commented:
> > @devabhishekpal In Overview Page In Route.json entries are also missing for local run getting 404 in local.
> > ![image](https://private-user-images.githubusercontent.com/112169209/343683342-199a48d3-9c05-4885-becd-d7b37a67bde5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk0ODcxODMsIm5iZiI6MTcxOTQ4Njg4MywicGF0aCI6Ii8xMTIxNjkyMDkvMzQzNjgzMzQyLTE5OWE0OGQzLTljMDUtNDg4NS1iZWNkLWQ3YjM3YTY3YmRlNS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyN1QxMTE0NDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT00YjE0ZjljZmI0NDAwMTA2YmFmZTY2MTdhYmFkZjgyZDc4ZmZjOTQzYTk2ZjcwZTk2NmUyNzQwZmNjMzQ0NzYwJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.06X0DfzXVgc9vi7qaDCC-gN2kRpObEXtDhQxxmNIcK8)
> 
> Hi @smitajoshi12 , I checked the older routes.json as well. I mean from older branch (Ozone 1.4.0) it seems the path mapping to `/triggerdbsync/om` was never present, hence the 404 response. But if we test it on a cluster instead of in dev mode, the API is being called. So it is okay in built files.

No Issues closing this comment

devabhishekpal commented:
> @devabhishekpal
> 
> Looks like Container ID column alignment has been changed.
> 
> ![image](https://private-user-images.githubusercontent.com/112169209/343713861-c1717426-7b77-4bff-ab3c-abc428900d39.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk0ODcxODMsIm5iZiI6MTcxOTQ4Njg4MywicGF0aCI6Ii8xMTIxNjkyMDkvMzQzNzEzODYxLWMxNzE3NDI2LTdiNzctNGJmZi1hYjNjLWFiYzQyODkwMGQzOS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyN1QxMTE0NDNaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT03Y2Y5MTZhYmEzNGE1YzI3MjQzNjUyZWZjOWQ0Y2NmYmUyNTVhMGNkOTlhYzgyYzUyZGY2YmZmMGQ4MzVmZWJlJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.YDg0pooPeH_Blpv-Qw4dTnqFgdQaj0ojxl49ttWtrwk)

It seems this is the default behaviour with expandable rows.
This is also present in the existing code, I cross checked with the current code.
The following is built on master

<img width="371" alt="Screenshot 2024-06-27 at 17 16 46" src="https://github.com/apache/ozone/assets/43001336/c8d72e41-25af-4d4f-bc96-84a39e69aa10">

devabhishekpal commented:
> @devabhishekpal
> 
> Can we increase width and Height of Pi Chart align more to left side so can get visibility for all entities.
> 
> ![image](https://private-user-images.githubusercontent.com/112169209/343748325-6e9241d2-e6c1-4e75-a3ff-671ba153809a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTk0OTIzNDQsIm5iZiI6MTcxOTQ5MjA0NCwicGF0aCI6Ii8xMTIxNjkyMDkvMzQzNzQ4MzI1LTZlOTI0MWQyLWU2YzEtNGU3NS1hM2ZmLTY3MWJhMTUzODA5YS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNjI3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDYyN1QxMjQwNDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1iMDRkNTQ4Y2ZiNDQxMGM5ZDhjYzNiNDM5MTBjYjcyYmIwOWQ3MWZjZjg3MjFiMDkxNzVlMjI5ZTUwYThlNmRkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.NLGj-Ha0dNVQywRt28-lSviDvI7xbB0w9m710_gxHmE)

So even if we increase the pie-chart width the percentage of entity area will be the same.
I added the changes to spread out the pie-chart and added padding to the Tab panels as you suggested.

However with a larger pie-chart the labels will overlap on the chart as seen below.
<img width="1725" alt="Screenshot 2024-06-27 at 18 37 10" src="https://github.com/apache/ozone/assets/43001336/4422991a-0cb7-4cf5-a179-0688fa5c7538">

So sticking to current size. I believe your PR to add a normalized width to the small entities would solve this issue of small areas.

smitajoshi12 commented:
It looks good to me now.

devabhishekpal commented on hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/.pnpmfile.cjs:
Removed the file in the latest commit

smitajoshi12 commented:
@devabhishekpal 

Response and Pi Chart are different

![image](https://github.com/apache/ozone/assets/112169209/3e110b5b-7cb3-45b8-ae2b-25595d136312)

Response:-
{
  "status": "OK",
  "path": "/vol1/bucket1/dir1/key1",
  "size": 3800,
  "sizeWithReplica": 11400,
  "subPathCount": 0,
  "subPaths": [],
  "sizeDirectKey": 0
}

smitajoshi12 commented:
> @devabhishekpal
> 
> Response and Pi Chart are different
> 
> ![image](https://private-user-images.githubusercontent.com/112169209/346509064-3e110b5b-7cb3-45b8-ae2b-25595d136312.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjA0MzE3ODgsIm5iZiI6MTcyMDQzMTQ4OCwicGF0aCI6Ii8xMTIxNjkyMDkvMzQ2NTA5MDY0LTNlMTEwYjViLTdjYjMtNDViOC1hZTJiLTI1NTk1ZDEzNjMxMi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwNzA4JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDcwOFQwOTM4MDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01NjM4NDljYjk1YjJmYzUxZWY0OWE5MjRiYTNhNjMxNGQ0NzNlOTZiYjgwOWMxOGVhY2MzNzdlYjA4ODhiNzk3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.rFbPT9v1CCs5baltm3l4uf7NH4CXvRckfhCBH4FrPag)
> 
> Response:- { "status": "OK", "path": "/vol1/bucket1/dir1/key1", "size": 3800, "sizeWithReplica": 11400, "subPathCount": 0, "subPaths": [], "sizeDirectKey": 0 }

It  looks Good to me after your recent changes

devabhishekpal commented:
Hi @ArafatKhan2198, yes - the changes have been tested with the the datasets that were used by @smitajoshi12 in the original normalization PR and it is working correctly.
The commit https://github.com/apache/ozone/pull/6841/commits/d8cdd9214036467ca91341cf369eadf1c3c83e50 adds the changes for the normalization.

ArafatKhan2198 reviewed (approved):
Thanks for the patch @devabhishekpal 
LGTM +1 
Thanks for improving the UI for recon :)

devmadhuu reviewed (approved):
Thanks @devabhishekpal . New changes LGTM +1

ArafatKhan2198 commented:
Thanks for the work @devabhishekpal & thanks @devmadhuu @smitajoshi12 @ivandika3 @adoroszlai for the review.
